### PR TITLE
src/components/profile: add ProfileDetails section with components for displaying settings

### PR DESCRIPTION
**Pull request by maceksimon**
Add components to render the Personal Details section on Profile page.

Composed primarily from 2 components: `ProfileDetails` + `DetailsItem`.
`ProfileDetails` lists data rows.
`DetailsItem` provides a wrapper for displaying a flexible content (personal details) and edit them by using a slotted form.

Adds edit forms for the first few rows following a common pattern.
Language switcher will probably be replaced with a more fitting design later.

Screenshot 2024-08-22 at 08 46 19

### CODE REVIEW

File: src/components/form/FormUpdateGender.vue [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+      required: true,
+    },
+  },
+  emits: ['update:value', 'close'],
+  setup(props, { emit }) {
+    const inputValue = ref<string>('');
+
+    const genderOptions: FormOption[] = [
+      {
+        label: 'Female',
```
Comment (tmszi) [inline src/components/form/FormUpdateGender.vue:54]: String should be translated.

File: src/components/form/FormUpdateGender.vue [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+        value: 'female',
```
Comment (tmszi) [inline src/components/form/FormUpdateGender.vue:55]: You can reuse existed Gender enum.

File: src/components/form/FormUpdateGender.vue (hunk 1 above)
Comment (maceksimon) [reply src/components/form/FormUpdateGender.vue]: Translated.

File: src/components/form/FormUpdateGender.vue (hunk 2 above)
Comment (maceksimon) [reply src/components/form/FormUpdateGender.vue]: Refactored using Gender enum.

**Review by tmszi — APPROVED** [review]

**Pull request merged**